src/middlewared/middlewared/plugins/replication.py

```diff
@@ -280,12 +280,44 @@ class Config:
     )
     async def do_delete(self, id):
 
+        replication = await self._get_instance(id)
+
+        try:
+            if replication['lastsnapshot']:
+                zfsname = replication['lastsnapshot'].split('@')[0]
+                await self.middleware.call('notifier.zfs_dataset_release_snapshots', zfsname, True)
+        except Exception:
+            pass
+        if os.path.exists(REPL_RESULTFILE):
+            with open(REPL_RESULTFILE, 'rb') as f:
+                data = f.read()
+            try:
+                results = pickle.loads(data)
+                results.pop(id, None)
+                with open(REPL_RESULTFILE, 'wb') as f:
+                    f.write(pickle.dumps(results))
+            except Exception as e:
+                self.logger.debug('Failed to remove replication from state file %s', e)
```

Comment:
There are a few file operations here that are not asyncio-friendly:
2 file opens, 1 unlink, 1 file read, 1 file write, 1 pickle load, 1 pickle dump
Is this enough to grant turning this coroutine into a method?

Reply:
Fixed